docs/revsets.md

```diff
@@ -219,6 +219,10 @@ for a comprehensive list.
 
 * `immutable_heads()`: Resolves to `trunk() | tags()` by default. See
   [here](config.md#set-of-immutable-commits) for details.
+* `immutable()`: Resolves to `::(immutable_heads() | root())` by default. See
+  [here](config.md#set-of-immutable-commits) for details.
+* `mutable()`: Resolves to `~immutable()` by default. See
+  [here](config.md#set-of-immutable-commits) for details.
```

Comment:
Maybe we should document that `immutable()`/`mutable()` aren't supposed to be customized?

iirc, there were concerns that user might override `immutable()` instead of `immutable_heads()` assuming it would affect whether commits can be rewritten.

Reply:
I've moved the documentation from the aliases section to the builtin functions section, since semantically an alias can be overridden, and I added a warning there not to modify it.

Reply:
> I've moved the documentation from the aliases section to the builtin functions section, since semantically an alias can be overridden

I understand aliases are equivalent to functions, but they are rendered as aliases in e.g. `jj config list`, so I prefer not to mix them in functions. (Functions can be shadowed by aliases anyway.)

Other than that, the doc change looks good, thanks.